src/v/cluster/distributed_kv_stm.h

```diff
@@ -78,6 +78,7 @@ requires std::is_trivially_copyable_v<Key>
          && std::is_trivially_copyable_v<Value>
 class distributed_kv_stm final : public raft::persisted_stm<> {
 public:
+    static constexpr const char* name = "distributed_kv_stm";
```

Comment:
Can we template this so that multiple versions can be made? (also should this be `std::string_view`)?